src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -7,8 +7,32 @@
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException(
+                    "Login cannot be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException(
+                    "User with this login already exists");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException(
+                    "Login must be at least 6 characters long");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException(
+                    "Password must be at least 6 characters long");
+        }
+        if (user.getAge() < 18) {
```

Comment:
You have a check for the minimum age of the user, which is good. However, there is no check for the maximum age of the user. It's always a good idea to limit the range of an integer to prevent very large numbers from being stored and to protect against attacks.